sql/plan/trigger.go

```diff
@@ -93,14 +93,42 @@ type triggerIter struct {
 	ctx            *sql.Context
 }
 
+// prependRowInPlanForTriggerExecution returns a transformation function that prepends the row given to any row source in a query
+// plan. Any source of rows, as well as any node that alters the schema of its children, will be wrapped so that its
+// result rows are prepended with the row given.
+func prependRowInPlanForTriggerExecution(row sql.Row) func(n, parent sql.Node, childNum int) (sql.Node, error) {
+	return func(n, parent sql.Node, childNum int) (sql.Node, error) {
+		switch n := n.(type) {
+		case *Project:
+			// Only prepend rows for projects that aren't the input to inserts and other triggers
+			switch parent.(type) {
+			case *InsertInto, *TriggerExecutor:
+				return n, nil
+			default:
+				return &prependNode{
+					UnaryNode: UnaryNode{Child: n},
+					row:       row,
+				}, nil
+			}
+		case *ResolvedTable, *IndexedTableAccess:
+			return &prependNode{
+				UnaryNode: UnaryNode{Child: n},
+				row:       row,
+			}, nil
+		default:
+			return n, nil
+		}
+	}
+}
+
```

Comment:
what's the difference between 
1) conditionally prepending the child row, as you do here
2) always prepending, and manipulating the parent expressions to access the correct fields of the composite row

Reply:
They're coupled. The parent expressions get their indexes from the analyzer using the scope, which is basically a promise that we'll fill in exactly the right number of outers scope columns at the beginning of the row at runtime. The issue is that for inserts, we don't ever consider the scope value, so we can't prepend any values to the projection.

This is complicated and there are still probably bugs in this approach.